packages/jest-runtime/src/script_transformer.js

```diff
@@ -295,16 +297,20 @@ export default class ScriptTransformer {
     let sourceMapPath: ?string = null;
 
     const willTransform =
-      !isInternalModule &&
-      !isCoreModule &&
-      (shouldTransform(filename, this._config) || instrument);
+      options.isRequiredByModuleField ||
+      (
+        !isInternalModule &&
+        !isCoreModule &&
+        (shouldTransform(filename, this._config) || instrument)
+      );
 
     try {
       if (willTransform) {
         const transformedSource = this.transformSource(
           filename,
           content,
           instrument,
+          options.isRequiredByModuleField,
```

Comment:
For some reason, `shouldTransform` is called twice, I just passed it through for now. Maybe someone from the jest team can check if it is actually necessary to call it twice.

Reply:
Twice within the same file or same code path?

Reply:
Same code path.